### ISSUE TITLE
Freemius SDK 2.6.0 causes fatal error in Site Editor when used with WP Multi Network plugin (in multisite)

### ISSUE THREAD

**Issue by codex-m** [Freemius/wordpress-sdk#667]
# 🐞 bug report

**Behavior**:
 -  `What is the issue? (*)`

In multisite using WP Multi Network plugin + any network activated plugin using Freemius SDK 2.6.0 - results to fatal error (white screen) on main site front end site editor.

STEPS TO REPRODUCE:

1. Setup a fresh blank WordPress multisite.
2. Enable [WordPress debug](https://wordpress.org/documentation/article/debugging-in-wordpress/#example-wp-config-php-for-debugging).
3. Login as network administrator.
4. While logged-in as super-admin, go to Main site homepage front end.
5. Click "Edit Site". Please see screenshot.
![edit-site](https://github.com/Freemius/wordpress-sdk/assets/8086122/46bd29eb-c35c-4251-9eda-6cadad3699e1)

6. Site editor loads and there is no white screen, etc. and no PHP fatal error notice.
7. Install [WP Multi Network plugin](https://wordpress.org/plugins/wp-multi-network/).
8. Network activate WP Multi Network Plugin.

9. Install any plugin that is powered by Freemius latest SDK 2.6.0. For example, let's install [Ivory Search plugin](https://wordpress.org/plugins/add-search-to-menu/) (current version 5.5.4 that uses Freemius SDK 2.6.0).

10. Network activate Ivory search plugin and skip opt-in to simplify activation.
11. Go back again to main site homepage front end and click again "Edit site" (see screenshot above to to where to click this).

**TEST RESULT**: I expected: Site editor to loads normally, instead I got white screen with the following errors below - please see stack trace.

IMPORTANT: Please take note that this affects any plugins using Freemius SDK 2.6.0 that is network activated and the site is also using WP Multi Network plugin. If the affected plugin is downgraded to use older version of Freemius SDK - it works. So basically this is a new bug in Freemius SDK 2.6.0. Originally reported here:

- https://wordpress.org/support/topic/prime-mover-1-9-2-plus-wp-multi-network-causes-fatal-error-in-site-editor/
- https://wordpress.org/support/topic/undefined-constant-secure_auth_cookie-3/

**Versions**: (*)
 -  `Freemius SDK Version:` 2.6.0
 -  `WordPress Version:` 6.3.2
 -  `PHP Version:` 8.1

**Plugin / Theme**: (*)
 -  `Name:`  Prime Mover plugin,  Ivory Search plugin or but this affects "any" plugin using Freemius SDK 2.6.0
 -  `Slug:`  add-search-to-menu 
 -  `Freemius ID:` 3826 (Prime Mover)

**Additional Information**:
 -  `Browser Type: (*)` Firefox
 -  `Browser Version: (*)` 119.0
 -  `OS: (*)` Ubuntu 20.04
 -  `Stack Traces:`
```
[07-Nov-2023 03:30:23 UTC] PHP Fatal error:  Uncaught Error: Undefined constant "AUTH_COOKIE" in /home/emerson/sourcecode/test100/wp-includes/pluggable.php:932
Stack trace:
#0 /home/emerson/sourcecode/test100/wp-includes/pluggable.php(694): wp_parse_auth_cookie()
#1 /home/emerson/sourcecode/test100/wp-includes/class-wp-hook.php(310): wp_validate_auth_cookie()
#2 /home/emerson/sourcecode/test100/wp-includes/plugin.php(205): WP_Hook->apply_filters()
#3 /home/emerson/sourcecode/test100/wp-includes/user.php(3622): apply_filters()
#4 /home/emerson/sourcecode/test100/wp-includes/pluggable.php(70): _wp_get_current_user()
#5 /home/emerson/sourcecode/test100/wp-includes/l10n.php(98): wp_get_current_user()
#6 /home/emerson/sourcecode/test100/wp-includes/l10n.php(152): get_user_locale()
#7 /home/emerson/sourcecode/test100/wp-includes/l10n.php(1300): determine_locale()
#8 /home/emerson/sourcecode/test100/wp-includes/l10n.php(1331): _load_textdomain_just_in_time()
#9 /home/emerson/sourcecode/test100/wp-includes/l10n.php(185): get_translations_for_domain()
#10 /home/emerson/sourcecode/test100/wp-includes/l10n.php(330): translate()
#11 /home/emerson/sourcecode/test100/wp-content/plugins/wp-multi-network/wp-multi-network/includes/classes/class-wp-ms-networks-admin.php(176): esc_html__()
#12 /home/emerson/sourcecode/test100/wp-content/plugins/wp-multi-network/wp-multi-network/includes/classes/class-wp-ms-networks-admin.php(35): WP_MS_Networks_Admin->set_feedback_strings()
#13 /home/emerson/sourcecode/test100/wp-content/plugins/wp-multi-network/wpmn-loader.php(159): WP_MS_Networks_Admin->__construct()
#14 /home/emerson/sourcecode/test100/wp-content/plugins/wp-multi-network/wpmn-loader.php(106): WPMN_Loader->includes()
#15 /home/emerson/sourcecode/test100/wp-content/plugins/wp-multi-network/wpmn-loader.php(218): WPMN_Loader->__construct()
#16 /home/emerson/sourcecode/test100/wp-content/plugins/wp-multi-network/wpmn-loader.php(190): wpmn()
#17 /home/emerson/sourcecode/test100/wp-includes/class-wp-hook.php(310): setup_multi_network()
#18 /home/emerson/sourcecode/test100/wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters()
#19 /home/emerson/sourcecode/test100/wp-includes/plugin.php(517): WP_Hook->do_action()
#20 /home/emerson/sourcecode/test100/wp-settings.php(427): do_action()
#21 /home/emerson/sourcecode/test100/wp-config.php(101): require_once('...')
#22 /home/emerson/sourcecode/test100/wp-load.php(50): require_once('...')
#23 /home/emerson/sourcecode/test100/wp-admin/admin.php(34): require_once('...')
#24 /home/emerson/sourcecode/test100/wp-admin/site-editor.php(12): require_once('...')
#25 {main}
  thrown in /home/emerson/sourcecode/test100/wp-includes/pluggable.php on line 932
[07-Nov-2023 03:30:23 UTC] PHP Fatal error:  Uncaught Error: Undefined constant "AUTH_COOKIE" in /home/emerson/sourcecode/test100/wp-includes/pluggable.php:932
Stack trace:
#0 /home/emerson/sourcecode/test100/wp-includes/pluggable.php(694): wp_parse_auth_cookie()
#1 /home/emerson/sourcecode/test100/wp-includes/class-wp-hook.php(310): wp_validate_auth_cookie()
#2 /home/emerson/sourcecode/test100/wp-includes/plugin.php(205): WP_Hook->apply_filters()
#3 /home/emerson/sourcecode/test100/wp-includes/user.php(3622): apply_filters()
#4 /home/emerson/sourcecode/test100/wp-includes/pluggable.php(70): _wp_get_current_user()
#5 /home/emerson/sourcecode/test100/wp-includes/l10n.php(98): wp_get_current_user()
#6 /home/emerson/sourcecode/test100/wp-includes/l10n.php(152): get_user_locale()
#7 /home/emerson/sourcecode/test100/wp-includes/l10n.php(894): determine_locale()
#8 /home/emerson/sourcecode/test100/wp-includes/class-wp-fatal-error-handler.php(47): load_default_textdomain()
#9 [internal function]: WP_Fatal_Error_Handler->handle()
#10 {main}
  thrown in /home/emerson/sourcecode/test100/wp-includes/pluggable.php on line 932
```
----
**Note:** `(*)` indicates required information. Without this information, your issue may be auto-closed.

> You can find your Freemius SDK version at `/freemius/start.php`

> You can find your Freemius product ID in the Freemius dashboard: `SETTINGS -> INTEGRATION`.

> Do not modify the titles or questions. Simply add your responses to the ends of the questions.
  Add more lines if needed.

**Comment by TobiasBg** [comment]
I also have a report on this issue: https://wordpress.org/support/topic/2-2-error-if-enter-edit-site/

In my case, WP Multi Network is not needed, but Wordfence needs to be network active (in addition to my plugin, TablePress, which uses the SDK 2.6.0 as well).

**Comment by bfintal** [comment]
Also got a report on this: https://wordpress.org/support/topic/fatal-error-with-secure_auth_cookie/

WP 6.4.1 Multisite
PHP 8.2

I think this should be prioritized

**Comment by swashata** [comment]
Hello @bfintal, @codex-m, @TobiasBg, we have prepared a small patch for this.

https://github.com/Freemius/wordpress-sdk/pull/669

Could you please help us with the patch by testing it against your environment? Thanks.

**Comment by bfintal** [comment]
> Could you please help us with the patch by testing it against your environment? Thanks.

Our QA has verified this, thanks!

**Comment by codex-m** [comment]
@swashata Thanks for the fix. I've tested this and it now works with WP 6.3.2 to WP 6.4.1 for which previously it does not work.

However the code comment in your fix specifies that the [temporary fix is only needed for WP versions 6.3.0 and 6.3.1](https://github.com/Freemius/wordpress-sdk/pull/669/commits/59cf9c69c649b4f11109d92f7d577322c66c44ee#diff-b455171d4fe2068f20af919c3e66f79459f0bc7f1dd81924715ce3bd0e78f6ccR55). Would it be better if we use this?

```
if (version_compare($version, "6.3", ">=") && version_compare($version, "6.3.1", "<="))`
```
instead of (the one you are using right now):

```
if (version_compare( $wp_version, '6.3.2', '<' )) {`
```

This will ensure the custom code only runs in affected versions WP 6.3 and WP 6.3.1 and to minimize it's side effects on all other WordPress versions. 

Just to take note that this issue reported is still reproducible  (as tested) even below WP 6.3 (using Freemius SDK 2.6.0.1 hotfix) but not if the plugin uses older Freemius SDK like version 2.5.12. So therefore we suggest to limit this change only to affected WP versions (WP 6.3 and WP 6.3.1).

Just a suggestion. Thanks again!

**Comment by swashata** [comment]
Thanks @codex-m, I initially did not put the inclusive check because I thought `site-editor.php` and `$_GET['wp_theme_preview']` was only available from WP 6.3+. But if your test is confirming that is not the case, then we will definitely include it.

**Comment by swashata** [comment]
@codex-m the PR has been updated, thanks.

**Comment by codex-m** [comment]
Thanks @swashata , looks good. Once a new Freemius SDK will be officially released with this fix - we will update our plugins and release a new version also.

**Comment by TobiasBg** [comment]
Hi @swashata,

thanks a lot! I can also confirm that the PR fixes the issue :-)

Best wishes,
Tobias

**Comment by swashata** [comment]
This has been resolved in v2.6.1 https://github.com/Freemius/wordpress-sdk/releases/tag/2.6.1 Thank you everyone 🙌
